controllers/eventingauth_controller.go

```diff
@@ -52,14 +52,17 @@ type eventingAuthReconciler struct {
 	errorRequeuePeriod   time.Duration
 	defaultRequeuePeriod time.Duration
 	iasClient            ias.Client
+	// existingIasApplications stores existing IAS apps in memory not to recreate again if exists
+	existingIasApplications map[string]ias.Application
```

Comment:
I think we shouldn't implement it like that, since we introduce a state and this will only be removed on application deletion. Also a handling like this will mean the objects in the map will grow over time and might not be released for a long time. It also means that we store all secrets in memory, which I think is a bad practice in terms of security.

Reply:
* Regarding, your first point of storing credentials in memory we already do something like this [here](https://github.com/kyma-project/kyma/blob/c2b2eb1a49e5c158f0e27373135e8f79ae894b35/components/eventing-controller/controllers/backend/reconciler.go#L91). I think it is really hard to hack the container memory as there are multiple security layers (container security, cluster security, etc.)
* Regarding memory usage we somethig similar with eventing subscriptions [here](https://github.com/kyma-project/kyma/blob/c2b2eb1a49e5c158f0e27373135e8f79ae894b35/components/eventing-controller/pkg/backend/jetstream/jetstream.go#L48) and I haven't heard anyone complaining about our eventing controller memory usage.